### PR TITLE
APEXCORE-625 log exception from eventloop.stop, and let original exception to propagate to caller.

### PR DESCRIPTION
**Pull request by tushargosavi**

### CODE REVIEW

**Review by deepak-narkhede — COMMENTED** [review]
@tushargosavi . Would it be helpful if the same is added for individual containers ( StreamingContainer.java line: 588) apart from Stram service.

**Comment by vrozov** [comment]
Please see my comments in JIRA

**Comment by tushargosavi** [comment]
@deepak-narkhede have added similar exception handling in StreamingContainer main.

**Comment by vrozov** [comment]
More comments added to JIRA. IMO, it is necessary to wait for the eventloop to stop, just handling an exception is not good enough, especially that with the latest netlet the exception will not be thrown.

File: engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java [hunk 1]
```diff
@@ -310,7 +310,13 @@ public static void main(String[] args) throws Throwable
         stramChild.heartbeatLoop();
         exitStatus = 0;
       } finally {
-        stramChild.teardown();
+        try {
+          stramChild.teardown();
+        } catch (Exception ex) {
+          if (exitStatus == 0) {
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java:316]: IMO, that is confusing. If an exception is thrown during teardown(), if exitStatus is not 0, the exception is swallowed, making it hard to debug JVM termination. If it is 0, it leads to JVM exiting with exit status 0, but logging exception both to AM and the container log.

**Comment by vrozov** [comment]
@tushargosavi closed due to inactivity

**Pull request closed without merging**